homeassistant/components/starline/__init__.py

```diff
@@ -46,10 +50,18 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     async def async_set_scan_interval(call):
         """Service for set scan interval."""
         options = dict(config_entry.options)
-        options[CONF_SCAN_INTERVAL] = call.data[CONF_SCAN_INTERVAL]
+        if call.service == SERVICE_SET_SCAN_INTERVAL:
```

Comment:
Rather than having two services registered to the same function, and then comparing the name of the called service to determine what action to take, would it make more sense to register a different function for each service? There is only one line different between the two functions, but that's 33% of the function.